src/components/views/avatars/MemberAvatar.tsx

```diff
@@ -69,20 +72,27 @@ export default class MemberAvatar extends React.Component<IProps, IState> {
     }
 
     private static getState(props: IProps): IState {
-        if (props.member?.name) {
+        let member = props.member;
+        if (member && !props.forceHistorical && SettingsStore.getValue("feature_use_only_current_profiles")) {
+            const room = MatrixClientPeg.get().getRoom(member.roomId);
+            if (room) {
+                member = room.getMember(MatrixClientPeg.get().getUserId());
```

Comment:
Doesn't this set everyone's profile to the current user?

Reply:
🤦 yep - good spot. It should be using the member's user ID.